### ISSUE TITLE
[VS Code Extension] Transit Nodes gives `turbo:no-such-task` error

### ISSUE THREAD

**Issue by jeniabrook** [vercel/turborepo#8577]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

n/a

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Which canary version will you have in your reproduction?

n/a

### Describe the Bug

Setting a transit node in the `turbo.json` file:
```json
"tasks": {
  "topology": {
    "dependsOn": ["^topology"]
  }
}
```
Raises an error: 
```
The task `topology` does not exist. (turbo:no-such-task)
```
**Screenshot from VS Code:**
![Screenshot 2024-06-23 at 18 57 26](https://github.com/vercel/turbo/assets/11990241/e4d9e8d1-2747-40e2-af9a-a092dc6d28cd)

### Expected Behavior

Add a whitelist for the extension that will ignore those special tasks:
```json
"turbo.transitNodes": ["topology"]
```

Or maybe to detect it automatically (not sure if possible).

### To Reproduce

1. Install Turborepo LSP v1.1.3
2. Install `turborepo` v2.0.4
3. Create in `turbo.json` a transit node

**Comment by aryanprince** [comment]
I have the same issue with a Turborepo using pnpm workspaces on a Mac. The error makes sense cus Transit Nodes have to use the name of a script that does *not* exist anywhere in the project as mentioned [here](https://arc.net/l/quote/gmvvpzdo).

Hoping some sort of whitelist behavior could be added like @jeniabrook mentioned. 

Alternatively, it would be nice if we had a certain keyword like "transit" or "topo" as the Transit Node name for everyone - as it currently seems too 'vague' to have any random keyword as the Transit Node task name.

**Comment by mikestopcontinues** [comment]
I don't think this should be a linting rule at all. For example, I have a "deploy" task that runs all the necessary scripts to build a given app. But I have no "deploy" scripts. Turbo supports this behavior—it shouldn't mark it as an error.

**Comment by SohelIslamImran** [comment]
Same isssue

![Image](https://github.com/user-attachments/assets/74e286d7-0aa8-4f2b-a891-2ff632b94c45)

**Comment by yamcodes** [comment]
Same issue

<img width="653" height="194" alt="Image" src="https://github.com/user-attachments/assets/86dd96e2-c930-4cd9-a0ca-4d493feade5e" />

It's also a little odd that we can "Run" the task via the extension but it's not a "real" task

<img width="662" height="87" alt="Image" src="https://github.com/user-attachments/assets/807687da-c121-48ca-aed7-af3e01f99981" />

**Comment by ivancuric** [comment]
It's weird that this is still not fixed. I have actually thought that I _needed_ to have each turbo task defined as a package.json task.

**Comment by apetta** [comment]
This is quite annoying to deal with given the explicit mention and recommendation of `transit` nodes in the docs. Please add an exception for the "transit" task.
